src/VisualStudio/Core/Def/SymbolSearch/VisualStudioSymbolSearchService.ProgressService.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.SymbolSearch;
+using Microsoft.VisualStudio.TaskStatusCenter;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.SymbolSearch
+{
+    internal partial class VisualStudioSymbolSearchService : AbstractDelayStartedService, ISymbolSearchService
+    {
+        private class ProgressService : ISymbolSearchProgressService
+        {
+            private readonly object _gate;
+            private readonly IVsTaskStatusCenterService _taskCenterServiceOpt;
+
+            private TaskCompletionSource<bool> _taskCompletionSource;
+
+            public ProgressService(IVsTaskStatusCenterService vsTaskStatusCenterService)
+            {
+                _taskCenterServiceOpt = vsTaskStatusCenterService;
+            }
+
+            public Task OnDownloadFullDatabaseStartedAsync(string title)
```

Comment:
AFAICT - you are invoking this on a background thread.  Is that safe?  Can it be documented on ISymbolSearchProgressService?